File: javascript/examples/custom-observability/test-no-auto-init.ts
/**
 * Test: Verify that importing @langwatch/scenario does NOT auto-initialize OpenTelemetry.
 *
 * Before this fix, just importing the module with LANGWATCH_API_KEY set would
 * trigger setupObservability() and instrument all HTTP requests, middleware, etc.
 */
import { trace } from "@opentelemetry/api";

// Check the provider BEFORE importing scenario
const providerBefore = trace.getTracerProvider();
const providerNameBefore = providerBefore.constructor.name;

// Dynamically import scenario to test the side-effect
const scenario = await import("@langwatch/scenario");

// Check the provider AFTER importing scenario
const providerAfter = trace.getTracerProvider();
const providerNameAfter = providerAfter.constructor.name;

console.log(`Provider before import: ${providerNameBefore}`);
console.log(`Provider after import:  ${providerNameAfter}`);

if (providerNameBefore === providerNameAfter) {
  console.log(
    "\nPASS: Importing @langwatch/scenario did NOT auto-initialize OpenTelemetry"
  );
  console.log("   The global TracerProvider is unchanged.");
} else {
  console.error(
    "\nFAIL: Importing @langwatch/scenario auto-initialized OpenTelemetry!"
  );
  console.error(
    `   Provider changed from ${providerNameBefore} to ${providerNameAfter}`
  );
  process.exit(1);
}

File: python/examples/custom_observability/test_no_auto_init.py
"""Test: Verify that importing scenario does not auto-init OTel."""
import os

# Ensure no LangWatch data is sent
os.environ["LANGWATCH_API_KEY"] = ""


def main():
    from opentelemetry import trace
    from opentelemetry.sdk.trace import TracerProvider

    # Check the provider BEFORE importing scenario
    provider_before = trace.get_tracer_provider()
    print(f"Provider before import: {type(provider_before).__name__}")

    # Import scenario
    import scenario  # noqa: F401

    # Check after import
    provider_after = trace.get_tracer_provider()
    print(f"Provider after import: {type(provider_after).__name__}")

    # Verify no TracerProvider was set
    is_sdk_provider = isinstance(provider_after, TracerProvider)
    print(f"Is SDK TracerProvider: {is_sdk_provider}")

    if is_sdk_provider:
        print("\nFAIL: Importing scenario auto-initialized OTel")
        exit(1)

    print("\nPASS: No auto-initialization on import")
    exit(0)


if __name__ == "__main__":
    main()
